packages/core-data/src/selectors.ts

```diff
@@ -344,7 +344,7 @@ export const getEntityRecord = createSelector(
 				const field = fields[ f ].split( '.' );
 				let value = item;
 				field.forEach( ( fieldName ) => {
-					value = value[ fieldName ];
+					value = value?.[ fieldName ];
```

Comment:
Could be nice to abort early when `value` becomes `undefined`. Also, this can be a good use of `reduce`:
```js
const value = fields.reduce( ( v, f ) => v?.[ f ], item );
```
But the early abort probably requires a `for` loop with `break` anyway.

Reply:
I agree, but FWIW I opted for brevity here. And I think we no longer need to abort early since we're covering `undefined` with the optional chaining.